src/utils/num.rs

```diff
@@ -0,0 +1,20 @@
+pub trait IsEven {
+    fn is_even(self) -> bool;
+}
+
+impl IsEven for i32 {
+    fn is_even(self) -> bool {
+        self % 2 == 0
+    }
+}
+
+impl IsEven for usize {
+    fn is_even(self) -> bool {
+        self % 2 == 0
+    }
+}
+
+// Generic function that works for any type that implements IsEven
+pub fn is_even<T: IsEven>(num: T) -> bool {
+    num.is_even()
+}
```

Comment:
This feels really convoluted, but i guess it works alright.

You could just use % 2 == 0 in place and it would be really understandable.